semgrep-core/tests/js/metavar_import3.sgrep

```diff
@@ -0,0 +1,3 @@
+import { $X } from 'dependency';
+...
+$X(...);
```

Comment:
Nice! Does this work for anything pulled into the namespace here? In other words, if I have `import { func1, func2, func3 }` this will detect `func1(...)`, `func2(...)`, or `func3(...)`? I can imagine user's will want it either way. Is there a way to specify if I _only_ care about `func1(...)`. E.g. something like `import { ..., func1, ... } from 'dependency';`?

Reply:
import {a, b, c} from 'X' is transformed internally as import {a} from 'X'; import {b} from 'X';
so yes doing import {$X} from 'dependency' will have $X match in turn func1, func2, and func3.
You can also do import { func1 } from 'dependency' and it will also match, even if it was written as import { func2, func3, func1 }

Reply:
Ahh, great, thanks for the clarification!